CDCgov/prime-reportstream
# 10921 extract HL7 truncation logic into UP

This PR extracts out HL7 truncation logic from the Covid pipeline coded directly in the HL7 serializer out into it's own module. It also adds that functionality to the UP.

Test Steps:
1. `./gradlew test`
2. `./gradlew testIntegration`

E2E Test Steps
1. Update/add the following settings to organizations.yml
```
schemaName: "metadata/hl7_mapping/ORU_R01/ORU_R01-base"
truncateHDNamespaceIds: true
truncateHl7Fields: "OBX-23-1,ORC-21-1,OBR-16-1,OBR-16-3,ORC-12-1,ORC-12-3,PID-5-1,PID-5-2,PID-5-3"
```
2. Send the attached FHIR bundle to the Reports API:
[Simple Report.postman_collection.txt](https://github.com/CDCgov/prime-reportstream/files/11165604/Simple.Report.postman_collection.txt)
This sample contains all the fields that should be truncated and you can search for them by looking for "truncatethis"
3. Verify that the request was accepted
4. Open the generated HL7 file in the local sftp folder.
5. Verify all the fields were truncated by making sure the HL7 message doesn't include the text "truncatethis"
6. Run integration tests and make sure they pass

## Changes
- Pull out HL7 truncation logic into own module
- Call new module from Covid pipeline
- Call new module in UP

### Testing
